helios-services/src/main/java/com/spotify/helios/servicescommon/ServiceParser.java

```diff
@@ -230,6 +236,10 @@ public String getZooKeeperClusterId() {
     return options.getString(zooKeeperClusterId.getDest());
   }
 
+  public Boolean getNoZooKeeperRegistration() {
+    return fromNullable(options.getBoolean(noZooKeeperRegistrationArg.getDest())).or(true);
```

Comment:
I think this is why the tests are failing. Doesn't the true make it so the master never registers by default?

Part of the master registration is creating the appropriate root nodes in ZooKeeper. Not creating those nodes would explain why the SoloIT fails.